specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN 
+| A secure system SHOULD implement a HW RoT
+
+|===
+
+==== Authorized Boot
+
+Multiple models can be used to ensure a secure system can only run authorized software.
+
+See xref:chapter2.adoc#_authorized_software[authorized software].
 
 ==== Attestation
 
+Multiple models can be used to prove to a reliant party that a secure system is in a trustworthy state.
 
-=== Basic virtualized system
+See xref:chapter2.adoc#_attestable_services[attestable services].
 
-==== Overview
+==== Sealing
 
-==== Isolation model
+Multiple models can be used to protect assets if a system is not in a trustworthy state.
+
+See xref:chapter2.adoc#_sealing[sealing].
 
 ==== Device access control
 
-==== Sealing
+For the purpose of this specification, a device can be a logical device. A physical device can present one or more logical devices, each with its own (logical) control interface.
 
-==== Attestation
+Isolation guarantees provided to software also apply to device initiated transaction. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT NNN
+| IOPMP MUST be used to guarantee that devices assigned to lower privilege levels cannot access resources assigned to M-mode.
+
+| CAT_NNN
+| IOPMP and IOMMU MUST be used to enforce access rules for devices assigned to user applications or guests on a virtualized system.
+
+|===
+
+On a non-virtualized system, user devices can be managed by the OS which can enforce access rules for user applications.
+
+On a virtualized system, devices can be virtualized and assigned to guests by the hypervisor configuring MMU and IOMMU translation rules. 
+
+In either case, physical device access rules are always enforced by IOPMP, controlled by RoT FW in M mode, preventing unauthorized device access to M-mode resources.
 
+=== Debug and performance management
 
-=== Global Platforms TEE
+See xref:chapter2.adoc#_security_lifecycle[security lifecycle]. +
+See https://github.com/riscv-non-isa/riscv-external-debug-security[enhanced RISC-V external debug security]
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT NNN
+| External debug SHOULD be enabled separately for M-mode and non M-mode software.
+
+| CAT_NNN
+| External debug MUST only be enabled by HW RoT (M-mode external debug) or by FW RoT (non M-mode external debug).
+
+|===
+
+Enables recoverable external debug of non M-mode software. In turn enabling supply chain separation.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| Self-hosted debug MAY be used for debug of non M-mode software.
+
+| CAT_NNN
+| Self-hosted debug MUST only be enabled by a higher privileged component.
+
+|===
+
+For example, within normal domain an S-mode OS can enable self-hosted debug for a user application. Only M-mode firmware should enable self-hosted debug for the S-mode OS.
```

Comment:
```suggestion
For example, within a supervisor domain an S-mode OS can enable self-hosted debug for a user application. Only M-mode firmware should enable self-hosted debug for the S-mode OS.
```

Reply:
This example is more specific to the use case, so I made the text specific to the use case as well.